web_widget_image_webcam/static/src/js/webcam_widget.js

```diff
@@ -0,0 +1,132 @@
+/* global Webcam */
+/*
+    Copyright 2016 Siddharth Bhalgami <siddharth.bhalgami@gmail.com>
+    License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+*/
+odoo.define("web_widget_image_webcam.webcam_widget", function (require) {
+    "use strict";
+
+    var core = require("web.core");
+    var rpc = require("web.rpc");
+    var Dialog = require("web.Dialog");
+    const {patch} = require("@web/core/utils/patch");
+    const {ImageField} = require("@web/views/fields/image/image_field");
+
+    var _t = core._t;
+    var QWeb = core.qweb;
+
+    const getWebcamFlashFallbackModeConfig = rpc.query({
+        model: "ir.config_parameter",
+        method: "get_webcam_flash_fallback_mode_config",
+    });
+
+    patch(ImageField.prototype, "web_widget_image_webcam", {
+        async _setWebcamParams() {
+            // ::webcamjs:: < https://github.com/jhuckaby/webcamjs >
+            // Webcam: Set Custom Parameters
+            Webcam.set({
+                width: 320,
+                height: 240,
+                dest_width: 320,
+                dest_height: 240,
+                image_format: "jpeg",
+                jpeg_quality: 90,
+                force_flash: (await getWebcamFlashFallbackModeConfig) === "1",
+                fps: 45,
+                swfURL: "/web_widget_image_webcam/static/src/lib/webcam.swf",
+            });
+        },
+
+        onWebcamClicked() {
+            var self = this,
+                WebCamDialog = $(QWeb.render("WebCamDialog")),
+                img_data = false;
+
+            self._setWebcamParams();
```

Comment:
```suggestion
        async onWebcamClicked() {
            var self = this,
                WebCamDialog = $(QWeb.render("WebCamDialog")),
                img_data = false;

            await self._setWebcamParams();
```

Now, since `_setWebcamParams` is an async function, you'd probably want to await it too.
Otherwise the flow of execution will continue and the params would not be set.

In real life, this will probably never happen because the click will surely go after the read config promise has already been resolved (user action is probably slower than http response execution)

But if we want to be peeky, you should definitely await this.

If you do this, then it would be good to refactor this method a little bit to make use of the `await` keyword. Things you could achieve with it:
- Remove the `var self = this` hack (not needed in modern js -- if callbacks are defined as arrow functions)
- Convert callbacks to arrow functions
- Convert `then()` to `await`
- Possibly move the Dialog's click buttons callbacks to separate methods (ie: `onDialogTakeSnapshotButtonClicked`, `onDialogSaveButtonClicked`). This would make the function much shorter, and easy to maintain.


Anyway, this is all nitpicking.